seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -412,40 +452,191 @@ def test_create_project_from_workspace(self, mock_tempfile, mock_mv_file, mock_s
+        mock_datetime.now.side_effect = lambda: datetime(2021, 3, 1, 0, 0, 0)
+        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY_ADD_DATA))
+        self.assertEqual(response.status_code, 200)
+        mock_api_logger.error.assert_not_called()
+
+        mock_tempfile.assert_called_with(mode='wb', delete=False)
+        mock_tempfile.return_value.__enter__.return_value.write.assert_called_with(
+            b's\nHG00731\nHG00732\nHG00733\nHG00735\nNA19675\nNA19675_1\nNA19678\nNA19678\nNA20870\nNA20874')
+        mock_mv_file.assert_called_with(
+            TEMP_PATH,
+            'gs://seqr-datasets/v02/GRCh37/AnVIL_WES/{guid}/base/{guid}_ids.txt'.format(guid=PROJECT1_GUID),
+            user=self.manager_user
+        )
+
+        self._assert_dag_trigger_requests(responses, ADD_DATA_UPDATED_ANVIL_VARIABLES, ADD_DATA_UPDATE_DAG_TASKS_RESP,
+                                          10, PROJECT1_GUID)
+
+        self._assert_sent_slack_message(Project.objects.get(guid=PROJECT1_GUID), mock_slack, mock_send_email, "GRCh37")
+
+        # Test saving ID file exception
+        responses.calls.reset()
+        url = reverse(add_workspace_data, args=[PROJECT2_GUID])
+        mock_mv_file.side_effect = Exception('Something wrong while moving the ID file.')
+
+        # Test triggering dag exception
+        responses.replace(responses.GET,
+                          '{}/api/v1/dags/seqr_vcf_to_es_AnVIL_WES_v0.0.1/dagRuns'.format(MOCK_AIRFLOW_URL),
+                          json=DAG_RUNS_RUNNING)
+
+        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY_ADD_DATA2))
+        self.assertEqual(response.status_code, 200)
+        project = Project.objects.get(guid=PROJECT2_GUID)
+
+        self._assert_loading_req_exceptions(
+            mock_api_logger, mock_airtable_logger, mock_slack, mock_send_email, project, "GRCh37", PROJECT2_SAMPLES)
+
+        # Test logged in locally
+        remove_token(
+            self.manager_user)  # The user will look like having logged in locally after the access token is removed
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, f'/login/google-oauth2?next=/api/project/{project.guid}/add_workspace_data')
+
+    def _test_missing_required_fields(self, url, fields, workspace_name):
+        # Test missing required fields in the request body
+        response = self.client.post(url, content_type='application/json', data=json.dumps({}))
+        self.assertEqual(response.status_code, 400)
+        field_str = ', '.join(fields)
+        self.assertEqual(response.reason_phrase, f'Field(s) "{field_str}" are required')
+        self.mock_get_ws_access_level.assert_called_with(self.manager_user, TEST_WORKSPACE_NAMESPACE, workspace_name)
+
+    def _test_sample_data_error(self, mock_load_file, url):
+        """ Test parsing sample data error """
```

Comment:
you are inconsistently using docstrings and regular comments for these helper functions. In general, small helper functions are self documenting and don't need any sort of comment here at all. Unless there is information contained in a comment or docstring that adds more detail than the name of the helper function, it is better to take the comment out completely

Reply:
Okay